torch/linalg/__init__.py

```diff
@@ -140,3 +140,28 @@
     >>> LA.norm(m[0, :, :]), LA.norm(m[1, :, :])
     (tensor(3.7417), tensor(11.2250))
 """)
+
+tensorinv = _add_docstr(_linalg.linalg_tensorinv, r"""
+linalg.tensorinv(input, ind=2) -> Tensor
+
+Returns the inverse of a given tensor.
```

Comment:
I find this function really confusing.

Let's say it was accurately described as "Computes the inverse of :input:`attr`, a matrix." Then I would expect this to work:

```
a = np.random.randn(4, 4)
np.linalg.tensorinv(a)
```

but instead it yells at me:
```
LinAlgError: Last 2 dimensions of the array must be square
```

Instead I think the second sentence is more accurate and should replace the first one:

"Computes a tensor inv = tensorinv(input, ind) such that tensordot(inv, ind) == I_n, where I_n is the n-dimensional identity tensor."

That sentence will also also need to define what n is.

Then I would include a discussion of ind, since it's far from intuitive.

Reply:
Another idea would be to create a note describing the difference between this function and inv (currently inverse in PyTorch: https://pytorch.org/docs/master/generated/torch.inverse.html?highlight=inv#torch.inverse).

Reply:
NumPy does not have any checks for the condition specified in the docs: the input must satisfy `prod(a.shape[:ind]) == prod(a.shape[ind:])`. I'll add such check here.
So your example doesn't work because `np.prod(a.shape[:2])` is 9 while `np.prod(a.shape[2:])` is 1.0, so the input is not "square" in their definition.